### DISCUSSION TITLE
I am looking for documentation of all object events

### DISCUSSION THREAD

**Discussion by zmiju** [fabricjs/fabric.js#8691] [Q&A]
Where can I find list of all object events? I am especially interested in textbox events.

**Comment by asturur** [comment]
hi @zmiju, i think that the textbox offers the events as IText described in this file here:
https://github.com/fabricjs/fabric.js/blob/master/src/shapes/IText/IText.ts#L22-L31
On top of the basic object events that you find similarly in the Object files.

Does that answer your question?
Currently a type system is being built and if you try the current master you should be able to get the events by autocomplete.

**Reply by zmiju** [reply]
Thanks, it helped a lot :)
